gc/gc-iceberg/src/main/java/org/apache/iceberg/GCMetadataUtil.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.io.FileIO;
+import org.projectnessie.gc.iceberg.ExpireContentsProcedure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util class with same package as Iceberg because {@link ManifestFiles#open(ManifestFile, FileIO)}
+ * is package private.
+ */
+public final class GCMetadataUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GCMetadataUtil.class);
+
+  public static List<String> computeAllFiles(Snapshot snapshot, FileIO io) {
+    // New ManifestList file created after compaction will have new manifest files
+    // pointing to older data files with a marking (status) that it is deleted data files.
+    // So, skip those datafiles.
+    Predicate<ManifestEntry<?>> activeDataFilesPredicate =
+        entry -> entry.status() != ManifestEntry.Status.DELETED;
+
+    try {
+      Stream<String> dataFileStream =
+          snapshot.allManifests().stream()
```

Comment:
Does all this work with table-metadata v1 as well?

Reply:
Yes, as mentioned in the spec, this field is mandatory for both v1 and v2. So, it will be present always.

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/5889404/178730265-c0bcde74-d32e-4ca5-a23b-506cca2fd03e.png">

Reply:
The question was: "Does **all** this work..." - aka scenarios with v1 table metadata instead of v2!

Reply:
By default (current test cases) Iceberg uses the V1 format. So, test cases cover V1 already. 
I will also V2 test cases (for this I need to upgrade spark to 3.2 as spark3.2 started supporting merge on read deletes). I will do it now.